tests/foreman/cli/test_hostgroup.py

```diff
@@ -230,15 +230,11 @@ def test_negative_create_with_content_source(module_org):
 @pytest.mark.run_in_one_thread
 @pytest.mark.tier2
 def test_positive_update_hostgroup(
-    request,
```

Comment:
I think the PR title is a little misleading, Puppet is still supported and there are many other Puppet options available related to Puppet which are available when using `session_puppet_enabled_sat` for hostgroup creation.

From related test failure, it's failing due to `Error: Unrecognised option '--environment-id'`, I'm not sure if an environment is required for host group creation @vsedmik can you confirm this behavior? if not can we try removing this option for create command?

Reply:
In general, the original puppet-related `--environment` was deprecated in favour of `--puppet-environment` quite some time ago. I wonder it survived for so long.
When looking into a puppet-enabled SAT, I can see:
```
[root@sat ~]# hammer hostgroup create --help
Usage:
    hammer hostgroup create [OPTIONS]

Options:
...
 --puppet-ca-proxy VALUE                   Name of Puppet CA proxy
 --puppet-ca-proxy-id NUMBER               Puppet CA Capsule id
 --puppet-class[es|-ids] LIST              Names/Ids of associated Puppet classes
 --puppet-environment[-id] VALUE/NUMBER    Name/id of associated Puppet environment
 --puppet-proxy VALUE                      Name of Puppet proxy
 --puppet-proxy-id NUMBER                  Puppet Capsule id
...
```
So `environment-id` -> `puppet-environment-id` might fix it if it fails on HG creation time.
The question is if @pondrejk wants to narrow down the case to bare HostGroup without puppet, or keep the current full scenario with puppet-enabled SAT.

Reply:
Ok, whatabout a compromise, I'll keep the original puppet satellite-specific test and the new one as well

Reply:
@pondrejk I think instead of managing 2 similar tests for puppet and non-puppet, it would be nice to have parametrization for this and other tests in this module, similar @adarshdubey-star has implemented in https://github.com/SatelliteQE/robottelo/pull/10390 for Azure tests, non-blocking atm.